platform-sdk/swirlds-virtualmap/src/main/java/com/swirlds/virtualmap/internal/hash/AbstractHashHoldingTask.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (C) 2023-2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.virtualmap.internal.hash;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.concurrent.ForkJoinPool;
+import java.util.concurrent.ForkJoinTask;
+import java.util.concurrent.ForkJoinWorkerThread;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * A unit of work.
+ */
+public abstract class AbstractHashHoldingTask extends ForkJoinTask<Void> {
```

Comment:
It doesn't have to change as a part of this PR but the name looks a bit confusing. Even though it's called `AbstractHashHoldingTask`, it has nothing to do with a hash. I understand that it has `HashHoldingTask` as a child but this reason doesn't seem sufficient. Can it have more generic name?

Reply:
The wiring framework was built on top of the task parallelism framework represented by `AbstractTask` but should have never been considered its owner. `VirtualHasher` uses the same task parallelism framework and has no relation to the wiring framework.
This PR duplicates the task parallelism framework in two places. `AbstractTask` should remain in `com.swirlds.common` and be accessible by both. Further work on the task parallelism framework will extend it beyond a single class and should not maintain two copies of it.

Reply:
Going ahead and closing the PR, since we determined as a team that this is only worth doing if no additional complications arise.